unitTests/main.cpp

```diff
@@ -1997,62 +1884,62 @@ TEST_F(UnitType, unitTypeMixedUnitMultiplication)
-	EXPECT_NEAR(32.2917333168, e_ft2(), 5.0e-6);
-	e_ft2 = f * meter_t<double>(3);
-	EXPECT_NEAR(32.2917333168, e_ft2(), 5.0e-6);
-
-	auto mps = meter_t<double>(10.0) * unit<inverse<seconds>>(1.0);
-	EXPECT_EQ(mps, meters_per_second_t<double>(10));
-	mps = unit<meters, int>(10) * unit<inverse<seconds>, int>(1);
-	EXPECT_EQ(mps, meters_per_second_t<double>(10));
-	mps = meter_t<double>(10.0) * unit<inverse<seconds>, int>(1);
-	EXPECT_EQ(mps, meters_per_second_t<double>(10));
-	mps = unit<meters, int>(10) * unit<inverse<seconds>>(1.0);
-	EXPECT_EQ(mps, meters_per_second_t<double>(10));
+	c_m2 = b_ft * meter<double>(2);
+	EXPECT_NEAR(2.0, c_m2.value(), 5.0e-5);
+	c_m2 = f * meter<int>(2);
+	EXPECT_NEAR(2.0, c_m2.value(), 5.0e-5);
+	c_m2 = b_ft * meter<int>(2);
+	EXPECT_NEAR(2.0, c_m2.value(), 5.0e-5);
+	c_m2 = f * meter<double>(2);
+	EXPECT_NEAR(2.0, c_m2.value(), 5.0e-5);
+
+	unit<squared<foot_conversion_factor>> e_ft2 = b_ft * meter<double>(3);
+	EXPECT_NEAR(32.2917333168, e_ft2.value(), 5.0e-6);
+	e_ft2 = f * meter<int>(3);
+	EXPECT_NEAR(32.2917333168, e_ft2.value(), 5.0e-6);
+	e_ft2 = b_ft * meter<int>(3);
+	EXPECT_NEAR(32.2917333168, e_ft2.value(), 5.0e-6);
+	e_ft2 = f * meter<double>(3);
+	EXPECT_NEAR(32.2917333168, e_ft2.value(), 5.0e-6);
+
+	auto mps = meter<double>(10.0) * unit<inverse<second_conversion_factor>>(1.0);
+	EXPECT_EQ(mps, meters_per_second<double>(10));
+	mps = meter<int>(10) * unit<inverse<second_t>, int>(1);
+	EXPECT_EQ(mps, meters_per_second<double>(10));
+	mps = meter<double>(10.0) * unit<inverse<second_t>, int>(1);
+	EXPECT_EQ(mps, meters_per_second<double>(10));
+	mps = meter<int>(10) * unit<inverse<second_t>>(1.0);
+	EXPECT_EQ(mps, meters_per_second<double>(10));
 }
 
 TEST_F(UnitType, unitTypedimensionlessMultiplication)
 {
-	meter_t<double> a_m(1.0);
+	meter<double> a_m(1.0);
 
 	auto result_m = dimensionless<double>(3.0) * a_m;
-	EXPECT_NEAR(3.0, result_m(), 5.0e-5);
+	EXPECT_NEAR(3.0, result_m.value(), 5.0e-5);
 
 	result_m = a_m * dimensionless<double>(4.0);
-	EXPECT_NEAR(4.0, result_m(), 5.0e-5);
+	EXPECT_NEAR(4.0, result_m.value(), 5.0e-5);
 
 	result_m = 3.0 * a_m;
-	EXPECT_NEAR(3.0, result_m(), 5.0e-5);
+	EXPECT_NEAR(3.0, result_m.value(), 5.0e-5);
 
 	result_m = a_m * 4.0;
-	EXPECT_NEAR(4.0, result_m(), 5.0e-5);
+	EXPECT_NEAR(4.0, result_m.value(), 5.0e-5);
 
-	bool isSame = std::is_same_v<decltype(result_m), meter_t<double>>;
+	bool isSame = std::is_same_v<decltype(result_m), meter<double>>;
 	EXPECT_TRUE(isSame);
 }
 
 TEST_F(UnitType, unitTypeDivision)
 {
-	meter_t<double> a_m(1.0), b_m(2.0);
-	foot_t<double> a_ft(3.28084);
-	second_t<double> a_sec(10.0);
-	const unit<meters, int> d_m(1), e_m(2);
-	const std::common_type_t<unit<meters, int>, unit<feet, int>> j(d_m);
-	const unit<seconds, int> b_sec(10);
+	meter<double> a_m(1.0), b_m(2.0);
+	foot<double> a_ft(3.28084);
+	second<double> a_sec(10.0);
+	const meter<int> d_m(1), e_m(2);
+	const std::common_type_t<meter<int>, foot<int>> j(d_m);

```

Comment:
This is fine since the common type of `unit`s are tested above, but generally I don't think there were undesirable uses of `unit` left that weren't testing `unit` and not the strong types.

Reply:
should I revert c8d935f, or would you rather patch up the tests?

Reply:
Reverting seems the better option.